api/app/controllers/api/v1/parent_tasks_controller.rb

```diff
@@ -19,15 +20,23 @@ def create
             end
 
             def destroy
-                @parent_task.destroy
-                render json: { status: 'SUCCESS', message: 'Deleted the task', data: @parent_task }
+                if current_api_user.id == @parent_task.user_id
```

Comment:
ここは条件満たしてない場合に先にreturnしちゃうとネスト深くならなくてコードキレイキレイ

```suggestion
                return render json: { status: 'ERROR', message: 'You do not have permission'} unless current_api_user.id == @parent_task.user_id
                @parent_task.destroy
                render json: { status: 'SUCCESS', message: 'Deleted the task', data: @parent_task }
```